src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs

```diff
@@ -90,94 +92,68 @@ int IVsEditorFactory.Close()
                 }
             }
 
-            var editorAdaptersFactoryService = _componentModel.GetService<IVsEditorAdaptersFactoryService>();
-
             // Do we need to create a text buffer?
             if (textBuffer == null)
             {
-                var contentTypeRegistryService = _componentModel.GetService<IContentTypeRegistryService>();
-                var contentType = contentTypeRegistryService.GetContentType(ContentTypeName);
-                textBuffer = editorAdaptersFactoryService.CreateVsTextBufferAdapter(_oleServiceProvider, contentType);
-
-                if (_encoding)
-                {
-                    if (textBuffer is IVsUserData userData)
-                    {
-                        // The editor shims require that the boxed value when setting the PromptOnLoad flag is a uint
-                        var hresult = userData.SetData(
-                            VSConstants.VsTextBufferUserDataGuid.VsBufferEncodingPromptOnLoad_guid,
-                            (uint)__PROMPTONLOADFLAGS.codepagePrompt);
-
-                        if (ErrorHandler.Failed(hresult))
-                        {
-                            return hresult;
-                        }
-                    }
-                }
+                textBuffer = GetDocumentData(grfCreateDoc, pszMkDocument, vsHierarchy, itemid) as IVsTextBuffer;
```

Comment:
```suggestion
                textBuffer = (IVsTextBuffer)GetDocumentData(grfCreateDoc, pszMkDocument, vsHierarchy, itemid);
```

It seems if that's returning something else, we have bigger problems. The "as" will just mask the failure. I'm also still unsure how this can be null...

Reply:
Suggestion seems unresolved?